zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/utils/SearchEngineClientUtils.java

```diff
@@ -62,4 +63,27 @@ public static <T> T mapToSettings(
           e);
     }
   }
+
+  public static boolean allImportersCompleted(
+      final List<ImportPositionEntity> recordReaderStatuses, final int totalValueTypesCount) {
+    // For a fresh install where there are no import position documents
+    if (recordReaderStatuses.isEmpty()) {
+      return true;
+    }
+
+    final var partitionCompleted =
+        recordReaderStatuses.stream().anyMatch(ImportPositionEntity::getCompleted);
+
+    // If all record readers have an import position document we can check their completed values
+
+    if (recordReaderStatuses.size() == totalValueTypesCount) {
+      return recordReaderStatuses.stream().allMatch(ImportPositionEntity::getCompleted);
+    }
+
+    // If some record readers are missing their import position document we can say their status
+    // is equal to that of the partition completion status.
+
+    return recordReaderStatuses.stream().allMatch(ImportPositionEntity::getCompleted)
+        && partitionCompleted;
```

Comment:
🔧 This additional check is I think not necessary.

Reply:
refactored importers completed logic so this check is not needed